src/annotator/hypothesis-injector.js

```diff
@@ -85,13 +85,13 @@ export class HypothesisInjector {
 }
 
 /**
- * Check if the Hypothesis client has already been injected into an iframe
+ * Check if the client was added to a frame by {@link injectHypothesis}.
  *
  * @param {HTMLIFrameElement} iframe
  */
 function hasHypothesis(iframe) {
-  const iframeWindow = /** @type {Window} */ (iframe.contentWindow);
-  return '__hypothesis' in iframeWindow;
+  const iframeDocument = /** @type {Document} */ (iframe.contentDocument);
+  return iframeDocument.querySelector('script.js-hypothesis-config') !== null;
```

Comment:
Strictly, Hypothesis client could be injected without the need of a configuration (default values), isn't?

Wouldn't be better to check for the boot script?

Reply:
The problem with the boot script is that it can have several names:

* `.../build/boots.js`
* `https://cdn.hypothes.is/hypothesis`

Reply:
An alternative suggestion: check for `link[type="application/annotator+html"][rel="sidebar"]`. This element is created just before the boot script executes. We use this link to destroy the application (see 'src/annotator/index`), so maybe it is also good as a check if the application has started. It also is specific enought.

Reply:
> This element is created just before the boot script executes. 

This element is created _by_ the boot script. I think this would be a good suggestion except that it leads to a race condition - `hasHypothesis` will return true immediately after the boot script is injected. This is why the code currently looks for the configuration that is injected prior to the boot script. It is true that the client can be injected without any configuration, but in that case it will function as a host frame rather than a guest only frame.